src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentWorkflowNavigator.java

```diff
@@ -0,0 +1,72 @@
+package org.breedinginsight.brapps.importer.services.processors.experiment;
+
+import io.micronaut.context.annotation.Primary;
+import org.breedinginsight.brapps.importer.model.imports.ImportServiceContext;
+import org.breedinginsight.brapps.importer.model.workflow.ImportWorkflow;
+import org.breedinginsight.brapps.importer.model.workflow.ExperimentWorkflow;
+import org.breedinginsight.brapps.importer.model.workflow.ImportWorkflowResult;
+
+import javax.inject.Singleton;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Primary
+@Singleton
+public class ExperimentWorkflowNavigator implements ExperimentWorkflow {
+    private final List<ExperimentWorkflow> workflows;
+
+    public ExperimentWorkflowNavigator(List<ExperimentWorkflow> workflows) {
+        this.workflows = workflows;
+    }
+
```

Comment:
@dmeidlin @nickpalladino Is `workflows` injected in the constructor? If yes, how does that work? Usually we use the `@Inject` annotation, which is what I would have expected.

Reply:
ExperimentWorkflow is a @FunctionalInterface. Micronaut scans the code base and collects all instances of ExperimentWorkflow into an ordered list. This list is automatically available for constructor injection for the instance of ExperimentWorkflow annotated with @Primary.